File: docker_compose/feinplanung/HiCuMes.Persistence/Entities/Camundasubproductionstep.cs
using System;
using System.Collections.Generic;

namespace HiCuMes.Persistence.Entities;

public partial class Camundasubproductionstep
{
    public long Id { get; set; }

    public DateTime? CreateDateTime { get; set; }

    public string? ExternalId { get; set; }

    public DateTime? UpdateDateTime { get; set; }

    public string? VersionNr { get; set; }

    public string? CamundaProcessVariables { get; set; }

    public string? FilledFormField { get; set; }

    public string? FormField { get; set; }

    public string? FormKey { get; set; }

    public string? Name { get; set; }

    public string? TaskDefinitionKey { get; set; }

    public string? TaskId { get; set; }

    public long? CamundaMachineOccupationId { get; set; }

    public long? SubProductionStepId { get; set; }

    public  Camundamachineoccupation? CamundaMachineOccupation { get; set; }

    public  Subproductionstep? SubProductionStep { get; set; }
}

File: manufacturingTerminal/frontendManufacturingTerminal/src/app/interfaces/camunda-sub-production-step.ts
import {SubProductionStep} from "./sub-production-step";

export interface CamundaSubProductionStep {
  id: number
  externalId: string
  taskId: string
  name: string
  formKey: string
  taskDefinitionKey: string
  subProductionStep: SubProductionStep
  formField: [{
    type: string;
    required: string
    value: any
    key: string
    controlType: string
    label: string
    order: number
    options: [any]
  }]
}
